### DISCUSSION TITLE
`instance_key` as categorical in PointsModel

### DISCUSSION THREAD

**Discussion by giovp** [scverse/spatialdata#161] [Q&A]
Unrelated to the above but related to `instance_key`. I think that the column specified by `instance_key` should never be categorical, so we should remove this line: https://github.com/scverse/spatialdata/blob/4fb3433b7a188d462cbd3f250af26c4027023f89/spatialdata/_core/models.py#L464

_Originally posted by @LucaMarconato in https://github.com/scverse/spatialdata/issues/158#issuecomment-1448220704_

**Comment by giovp** [comment]
so in this case, `instance_key` refers to points, not shapes. I agree it's debatable on whether we want to have it categorical. I think it might make sense in e.g. few regions but many points. I think few can be order of thousands of regions (and millions of points). wdyt? @LucaMarconato

**Comment by LucaMarconato** [comment]
Ah you are right! I had confused points and circles (have been working with napari today and they use the same name). It should be categorical I agree.
